torch_geometric/nn/models/explainer.py

```diff
@@ -114,31 +114,265 @@ def forward(self, mask, *args):
+    See :obj:`torch_geometric.nn.to_captum_model` for example usage.
+
+    Args:
+
+        data (HeteroData or Data): The data to explain.
+        mask_type (str): Denotes the type of mask to be created with
+            a Captum explainer. Valid inputs are :obj:`"edge"`, :obj:`"node"`,
+            and :obj:`"node_and_edge"`:
+        *args: Additional forward arguments of the model being explained
+            which will be added to `additonal_forward_args`.
+            For :obj:`Data` this is arguments other than `x` and `edge_index`.
+            For :obj:`HeteroData` this is arguments other than `x_dict`
+            and `edge_index_dict`.
+    """
+    _raise_on_invalid_mask_type(mask_type)
+
+    additional_forward_args = []
+    if isinstance(data, Data):
+        if mask_type == "node":
+            inputs = [data.x.unsqueeze(0)]
+        elif mask_type == "edge":
+            inputs = [_to_edge_mask(data.edge_index).unsqueeze(0)]
+            additional_forward_args.append(data.x)
+        else:
+            inputs = [
+                data.x.unsqueeze(0),
+                _to_edge_mask(data.edge_index).unsqueeze(0)
+            ]
+        additional_forward_args.append(data.edge_index)
+
+    else:
+        metadata = data.metadata()
+        node_types = metadata[0]
+        edge_types = metadata[1]
+        inputs = []
+        if mask_type == "node":
+            for key in node_types:
+                inputs.append(data[key].x.unsqueeze(0))
+        elif mask_type == "edge":
+            for key in edge_types:
+                inputs.append(_to_edge_mask(data[key].edge_index).unsqueeze(0))
+            additional_forward_args.append(data.x_dict)
+        else:
+            for key in node_types:
+                inputs.append(data[key].x.unsqueeze(0))
+            for key in edge_types:
+                inputs.append(_to_edge_mask(data[key].edge_index).unsqueeze(0))
+        additional_forward_args.append(data.edge_index_dict)
+
+    additional_forward_args.extend(args)
+    return tuple(inputs), tuple(additional_forward_args)
+
+
+def captum_output_to_dicts(
+    captum_attrs: Tuple[Tensor], mask_type: str, metadata: Metadata
+) -> Tuple[Optional[Dict[NodeType, Tensor]], Optional[Dict[EdgeType, Tensor]]]:
+    r"""Convert the output of `Captum.ai <https://captum.ai/>`_ attribution
+    methods which is a tuple of attributions to two dictonaries with node and
+    edge attribution tensors. This function is used while explaining
+    :obj:`HeteroData` objects. See :obj:`torch_geometric.nn.to_captum_model`
+    for example usage.
+
+    Args:
+        captum_attrs (tuple[tensor]): The output of attribution methods.
+        mask_type (str): Denotes the type of mask to be created with
+            a Captum explainer. Valid inputs are :obj:`"edge"`, :obj:`"node"`,
+            and :obj:`"node_and_edge"`:
+
+            1. :obj:`"edge"`: Output will contain only edge attributions
```

Comment:
```suggestion
1. :obj:`"edge"`: :obj:`captum_attrs` contains only edge attributions. The output will contain a dictionary with key `EdgeType` and values

```

Reply:
I was confused at first because we use the word output in captum_attrs and here. So I thought we were also talking about the output of the attribution method here. Same applies for the other mask types.

Reply:
Good point.